application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,177 @@
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandInteractionEvent event) {
+        MessageChannel messageChannel = event.getChannel();
+        if (isChannelOnCooldown(messageChannel)) {
+            event
+                .reply("Please wait a bit, this command can only be used once per %d %s.".formatted(
+                        getRemainingTime(messageChannel),
+                        COOLDOWN_DURATION_UNIT.toString().toLowerCase(Locale.US)))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        channelIdToLastCommandInvocation.put(messageChannel.getIdLong(), Instant.now());
+
+        Optional<TextChannel> auditLogChannel = getChannel(event);
+        if (auditLogChannel.isEmpty()) {
+            return;
+        }
+
+        List<ModAuditLogWriter.Attachment> attachments = getAttachments(event);
+
+        String user = getName(event);
+        MessageAction message = auditLogChannel.orElseThrow().sendMessageEmbeds(messageEmbed(user));
+
+        message = buildAttachment(attachments, message);
+        message.queue();
+
+        event.reply("Thank you for contacting the moderators").setEphemeral(true).queue();
+
+    }
+
+    private int getRemainingTime(MessageChannel messageChannel) {
+        return COOLDOWN_DURATION_VALUE - getTimePassedSinceCommandCalled(messageChannel);
+    }
+
+    @NotNull
+    private String getName(@NotNull SlashCommandInteractionEvent event) {
+        String user = event.getUser().getAsMention();
+        boolean optionalMessage = event.getOption(OPTION_MESSAGE_PRIVATE).getAsBoolean();
+        if (optionalMessage) {
+            user = "Anonymous";
+        }
+        return user;
+    }
+
+    private MessageAction buildAttachment(List<ModAuditLogWriter.Attachment> attachments,
+            MessageAction message) {
+        for (ModAuditLogWriter.Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContentRaw(), attachment.name());
+        }
+        return message;
+    }
+
+    @NotNull
+    private MessageEmbed messageEmbed(String user) {
+        return new EmbedBuilder().setAuthor("Modmail Command invoked")
+            .setColor(Color.BLACK)
+            .setTitle("/modmail from '%s' ".formatted(user))
+            .build();
+    }
+
+    @NotNull
+    private List<ModAuditLogWriter.Attachment> getAttachments(
+            @NotNull SlashCommandInteractionEvent event) {
+        String content = event.getOption(MESSAGE).getAsString();
+        List<ModAuditLogWriter.Attachment> attachments = new ArrayList<>();
+        attachments.add(new ModAuditLogWriter.Attachment("content.md", content));
+        return attachments;
+    }
+
+    private @NotNull Optional<TextChannel> getChannel(@NotNull SlashCommandInteractionEvent event) {
+        long guildId = 1004371840245964851L;
```

Comment:
magic number? why is this hardcoded?

Reply:
> magic number? why is this hardcoded? what guild is this?

Magic number? its assigned to a variable. 

Its server guild

Reply:
1. that's still a magic number
2. it's not. the TJ guild is 272761734820003841
3. don't hardcode the guild number on this or else it's gonna end badly

Reply:
dont forget that the bot is used on multiple guilds, not just TJ. this has to go into the config. (similar to the already existing invite in the guild).

Reply:
> 1. that's still a magic number
How is it a magic number? It clearly defines what the number represents. 
> 2. it's not. the TJ guild is 272761734820003841
Oh, you are probably right, I might have copied my test server guild il deal with it now

Reply:
> dont forget that the bot is used on multiple guilds, not just TJ (for example our test server or your personal server). this has to go into the config. (similar to the already existing invite in the guild).
> 
> and since nobody wants to write IDs into a config, u might want to use a name pattern and do `jda.getGuilds()` or sth like that, picking the guild that matches the name.
Right, il get right on it

Reply:
Done